Libraries/Image/resolveAssetSource.js

```diff
@@ -47,9 +48,9 @@ function getDevServerURL() {
 function getOfflinePath() {
   if (_offlinePath === undefined) {
     var scriptURL = SourceCode.scriptURL;
-    var match = scriptURL && scriptURL.match(/^file:\/\/(\/.*\/)/);
+    var match = scriptURL && scriptURL.match(/^(file:\/\/)?(\/.*\/)/);
     if (match) {
-      _offlinePath = match[1];
+      _offlinePath = (Platform.OS === 'android' ? 'file://' : '') + match[2];
```

Comment:
@frantic, these are two separate issues. One is with loading an image given a uri, one is with reporting the scriptUrl with the "file://" prefix. Fixing one will not fix the other. The two issues in detail:

1. RN Android report its scriptURL without file:// prefix. This can be fixed easily. Note that we will still need the extra test cases to test for the scriptURL coming from assets:// vs file://, because the URI resolved for either case is different:

2. [RN Android needs a scheme prefix, otherwise it will assume that it is from the binary's static resources.](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java#L204) Therefore, this line of code is still needed in order to return an offline path prefixed with "file://" for Android, and no prefix for iOS. In iOS everything to specify a local asset all you need is a file path, but for Android, we are using Fresco and it relies on the scheme (assets://, file:// or res://) because all these resolve to different places, so I think it is better to rely on having this prefix encoded in the URI instead of complicated parsing to tell whether it is a resource ID (assets_submodule_logo), http location, assets location, or file location.